### PR TITLE
fix(ai-plugins): add missing kong.ctx.shared usage to llm.state

### PR DESCRIPTION
**Pull request by fffonion**
### Checklist

- [ ] The Pull Request has tests
- [ ] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

Missing from https://github.com/Kong/kong/pull/13155

### CODE REVIEW

**Review by oowl — APPROVED** [review]

File: kong/llm/drivers/shared.lua [hunk 1]
```diff
@@ -612,12 +612,12 @@ function _M.post_request(conf, response_object)
...
     local llm_latency = math.floor((ngx.now() - kong.ctx.plugin[start_time_key]) * 1000)
     request_analytics_plugin[log_entry_keys.META_CONTAINER][log_entry_keys.LLM_LATENCY] = llm_latency
-    kong.ctx.shared.ai_request_latency = llm_latency
+    llm_state.set_metrics("e2e_latency", llm_latency)
 
     if response_object.usage and response_object.usage.completion_tokens then
       local time_per_token = math.floor(llm_latency / response_object.usage.completion_tokens)
       request_analytics_plugin[log_entry_keys.USAGE_CONTAINER][log_entry_keys.TIME_PER_TOKEN] = time_per_token
-      kong.ctx.shared.ai_request_time_per_token = time_per_token
+      llm_state.set_metrics("totp_latency", time_per_token)
```
Comment (fffonion) [inline kong/llm/drivers/shared.lua:620]:
```suggestion
      llm_state.set_metrics("tpot_latency", time_per_token)
```

File: kong/llm/drivers/shared.lua [hunk 2]
```diff
@@ -612,12 +612,12 @@ function _M.post_request(conf, response_object)
   if kong.ctx.plugin[start_time_key] then
     local llm_latency = math.floor((ngx.now() - kong.ctx.plugin[start_time_key]) * 1000)
     request_analytics_plugin[log_entry_keys.META_CONTAINER][log_entry_keys.LLM_LATENCY] = llm_latency
-    kong.ctx.shared.ai_request_latency = llm_latency
+    llm_state.set_metrics("e2e_latency", llm_latency)
```
Comment (fffonion) [inline kong/llm/drivers/shared.lua:615]: this is only used in EE side,  see https://github.com/Kong/kong/pull/13472

File: kong/llm/state.lua [hunk 3]
```diff
@@ -94,4 +94,14 @@ function _M.get_response_tokens_count()
...
+  local m = kong.ctx.shared.llm_metrics or {}
+  m[key] = value
+  kong.ctx.shared.llm_metrics = m
+end
+
+function _M.get_metrics(key)
+  return (kong.ctx.shared.llm_metrics or {})[key]
+end
+
 return _M
```
Comment (StarlightIbuki) [inline kong/llm/state.lua:107]:
```suggestion
return _M

```

File: kong/llm/drivers/shared.lua [hunk 4]
```diff
@@ -612,12 +612,12 @@ function _M.post_request(conf, response_object)
...
 
     if response_object.usage and response_object.usage.completion_tokens then
       local time_per_token = math.floor(llm_latency / response_object.usage.completion_tokens)
       request_analytics_plugin[log_entry_keys.USAGE_CONTAINER][log_entry_keys.TIME_PER_TOKEN] = time_per_token
-      kong.ctx.shared.ai_request_time_per_token = time_per_token
+      llm_state.set_metrics("tpot_latency", time_per_token)
```
Comment (StarlightIbuki) [inline kong/llm/drivers/shared.lua:620]: I need some context for this renaming of fields

**Review by jschmid1 — APPROVED** [review]

File: kong/llm/drivers/shared.lua (hunk 4 above)
Comment (fffonion) [reply kong/llm/drivers/shared.lua:620]: this is to match the field name used by ai-proxy-advanced https://github.com/Kong/kong-ee/blob/master/kong/plugins/ai-proxy-advanced/schema.lua#L57

**Review by StarlightIbuki — APPROVED** [review]

**Comment by fffonion** [comment]
ce->ee cherrypick is handled in https://github.com/Kong/kong-ee/pull/9912

**Pull request merged**